src/bp/core/security/auth-service.ts

```diff
@@ -92,7 +92,10 @@ export class AuthService {
       return strategy && this._getStrategyConfig(strategy, strategyName)
     })
 
-    return { strategies: strategies.filter(Boolean), isFirstUser: await this.isFirstUser() }
+    const hiddenStrategies = strategies
+      .filter(strategy => config.authStrategies[strategy.strategyId].hidden)
+      .map(strategy => strategy.strategyId)
+    return { strategies: strategies.filter(Boolean), isFirstUser: await this.isFirstUser(), hiddenStrategies }
```

Comment:
FYI, if you change the logic of `this._getStrategyConfig` to add your field `hidden` to the config, you would no longer require the use of "hiddenStrategies", and you could simply filter on the strategies, ex: `data.payload.strategies.filter(x => !x.hidden)`